app/controllers/videos_controller.rb

```diff
@@ -1,8 +1,20 @@
 # frozen_string_literal: true
 
 class VideosController < ApplicationController
+  def index
+    authorize Video
+    return if search_params[:search].blank?
+    @videos = Video.search(search_params[:search])
+  end
```

Comment:
First I implemented this at a SearchController, then moved it here as it seemed more natural - as it would be possible to have it display all videos by default and then filter by the search query. But I don't know whether it's the best place or if there should be a `videos#search` action.

Reply:
I like your decision!